### PR TITLE
Use definitionsFullyResolved to look up consumes/produces + tests

### LINKED ISSUE #92: $ref in a path results in an Unknown Error
Looking at the swagger specification it seems that a `$ref` should be allowed in place of a path element, but this doesn't seem to be accepted by Sway.
For example, see the following YAML:

```

---
swagger: '2.0'
info:
  version: 1.0.0
  title: Test Api
  description: Test
schemes:
  - http
host: test.com
basePath: /api
paths:
  /my-actual-path/:
    $ref: '#/paths/x-mypath'

  x-mypath:
    get:
      description: Test
      responses:
        200:
          description: Response test
          schema:
            type: object
            properties:
              id:
                type: integer
```

This currently results in an Unknown Error.

### PR DESCRIPTION
**Pull request by lumaxis**
@whitlockjc This includes the changes from #101 plus I added tests.

For the tests, I basically duplicated the existing tests for `operation.js` but they are now run on both `swagger.yaml` and `swagger-relative-refs.yaml`.
Let me know if you'd prefer a different approach 👍 

Closes #101, fixes #92

### CODE REVIEW

File: test/helpers.js [hunk 1]
```diff
@@ -64,6 +66,24 @@ module.exports.getSwaggerApi = function (callback) {
   }
 };
 
+module.exports.getSwaggerApiRelativeRefs = function (callback) {
```
Comment (whitlockjc) [inline test/helpers.js]: Where is this used?  If it's unused, it should be removed from the PR.

Comment (lumaxis) [inline test/helpers.js]: See https://github.com/apigee-127/sway/pull/117/files#diff-a16611b9b020b32924e8d837c0994f79R39

Comment (whitlockjc) [inline test/helpers.js]: I guess I hadn't scrolled down enough.  I'll finish the review.

Comment (lumaxis) [inline test/helpers.js]: Yea, GitHub hides the diff of that file by default, you have to explicitly load it 🙂 
Also, just wanna let you know, that diff of that file looks way more complicated on GitHub than it actually is. It's really just copying all of the tests, indenting them one more level and changing them to use `swagger-relative-refs`.

File: test/test-operation.js [hunk 2]
```diff
@@ -32,260 +32,638 @@ var helpers = require('./helpers');
 var Sway = helpers.getSway();
 
 describe('Operation', function () {
-  var swaggerApi;
+  context('swaggerApiRelativeRefs', function () {
```
Comment (whitlockjc) [inline test/test-operation.js]: Any reason this is necessary?  In [previous tests](https://github.com/apigee-127/sway/blob/e53e529dc4a01b11a40f9e968fc15be76a292c73/test/test-issues.js#L69) where we use `./2.0/swagger-relative-refs.yaml` we just do it in that specific test.  Creating a `SwaggerApi` object using `Sway#create` with that file vs. using a helper is trivial so do we really need it?

_(If we decide "No", we could remove the helper completely.)_

Comment (lumaxis) [inline test/test-operation.js]: It's not strictly necessary but I added it because I [had to supply `jsonRefs` manually](https://github.com/apigee-127/sway/pull/117/files#diff-42a353e98c906af524c7d3d48feaf72aR75) to get the local refs parsed correctly.
Also, looking at [the test you linked](https://github.com/apigee-127/sway/blob/e53e529dc4a01b11a40f9e968fc15be76a292c73/test/test-issues.js#L69), I think this test is actually nor working. The result of `Swagger.create` appears to not be used at all and the test is really testing the object from [line 39](https://github.com/apigee-127/sway/blob/e53e529dc4a01b11a40f9e968fc15be76a292c73/test/test-issues.js#L39).

To sum it up, I'm happy to remove the helper but after considering the above, I'd suggest to use the helper in `test-issues.js` too.

Comment (whitlockjc) [inline test/test-operation.js]: Yeah, let's move it to `test-issues.js` and I'll merge it.  Having a helper makes sense.  Also, this is the only use of `context` as well so maybe we can do a per-test invocation of your helper?

Comment (lumaxis) [inline test/test-operation.js]: @whitlockjc Okay, I fixed the test in `test-issues.js` but I'm not sure I understand your question about `context`. `context` is just an alias of `describe` and I used it to group the two test groups for the "regular" `swaggerApi` and one using relative refs. Could you clarify what you meant?

screen shot 2016-12-23 at 00 31 16

Comment (whitlockjc) [inline test/test-operation.js]: I just meant that for consistency, it might be best to use `describe` instead of `context`.

Also, what editor are you using?

Comment (lumaxis) [inline test/test-operation.js]: That is VS Code with an Atom theme 🙂 
Sure, I can switch back to `describe` 👍 I just used it as I prefer `context` in my personal code because it reads a bit nicer.

Comment (whitlockjc) [inline test/test-operation.js]: I'm using the same setup, VS Code and the Atom One Dark theme.  As for `context` vs. `describe`, I think it's out of habit honestly.  _(`context` could be someone ambiguous as well.)_

Comment (lumaxis) [inline test/test-operation.js]: Alright, updated!

**Comment by whitlockjc** [comment]
Merged.
